content/en/platform/corda/5.0/deploying-operating/external-messaging/_index.md

```diff
@@ -0,0 +1,96 @@
+section = "corda.messaging"
+version = 1
+{
+    "config": "{
+        "routeDefaults": {
+            "active": false,
+            "inactiveResponseType": "IGNORE",
+            "receiveTopicPattern": "ext.$HOLDING_ID.$CHANNEL_NAME.receive"
+        }",
+    "schemaVersion": {
+        "major": 1,
+        "minor": 0
+    },
+    "section": "corda.messaging",
+    "version": 1
+})
+```
+{{% /tab %}}
+{{< /tabs >}}
+
+## Creating Kafka Topics
+
+Corda does not create Kafka topics used for external messaging. You must manually create a Kafka topic for each active and current route created for a virtual node. To determine the required topics and create these topics, do the following:
+
+1. Retrieve the configuration of the virtual vode. For more information, see [Retrieving Virtual Nodes]({{< relref "../vnodes/retrieving.md" >}}). 
+   If a CorDapp that implements external messaging is uploaded to a virtual node, the routes configuration is included in the response. For example:
+   ```json
+   {
+     "virtualNodes": [
+       {
+         ...
+         "externalMessagingRouteConfiguration": {
+           "currentRoutes": {
+             "cpiIdentifier": {
+               "name": "upgrade-testing-cordapp_ea6732c2-183a-4158-a9f0-7fe1084e8ee9",
+               "version": "v1",
+               "signerSummaryHash": "SHA-256:E53DA5C67A637D42335808FA1534005281BAE7E49CCE8833213E58E0FDCA8B35"
+             },
+             "routes": [
+               {
+                 "channelName": "external_app_v1",
+                 "externalReceiveTopicName": "ext.B5C16DBC7BB5.external_app_v1.receive",
+                 "active": true,
+                 "inactiveResponseType": "ERROR"
+               }
+             ]
+           },
+           "previousVersionRoutes": []
+         }
+       }
+     ]
+   }
+   ```
+   The `externalMessagingRouteConfiguration` field contains the configuration for the routes that were generated for each channel. The `currentRoutes` field specifies the current configuration while the `previousVersionRoutes` field contains historical configuration. The `previousVersionRoutes` field shows how the configuration has evolved over time, which can be useful when troubleshooting issues.
+
+2. Manually create a Kafka topic for each route returned in `externalMessagingRouteConfiguration.currentRoutes.routes`. In the example above, create a topic named `ext.B5C16DBC7BB5.external_app_v1.receive`. For more information, see the [Manual Bootstrapping section]({{< relref "../deployment/deploying/manual-bootstrapping.md">}}).
+
+3. Listen to the Kafka topic for messages using the `kafka-console-consumer` command. Login to the pod or container that is running Kafka and execute the following:
+   ```
+   kafka-console-consumer --topic <topic-nam> --from-beginning --bootstrap-server localhost:9092
+   ```
+   To test that everything is working correctly, start the flow that sends external messages using Swagger UI. 
```

Comment:
Do we have documentation that explains how to start a flow? It might be better to add a reference here.

Reply:
We don't have a proper section for this and I'm going to create a ticket (if we don't have one already) but in the meantime I've linked directly to the REST API docs.